model/src/main/proto/interaction_object.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Object type of the interaction object
+enum ObjectType {
+  NORMALIZED_STRING = 0;
+  SIGNED_INT = 1;
+  NON_NEGATIVE_INT = 2;
+  REAL = 3;
+  BOOLEAN = 4;
+  NUMBER_WITH_UNITS = 5;
+  SET_OF_HTML_STRING = 6;
+}
+
+// Structure for any interaction object
+message InteractionObject {
+  ObjectType object_type = 1;
+  oneof value {
+    string normalized_string = 2;
+    int32 signed_int = 3;
+    uint32 non_negative_int = 4;
+    float real = 5;
+    bool bool_value = 6;
+    NumberWithUnits number_with_units = 7;
+    // repeated fields cannot be within a oneof so lists are
+    // wrapped into this parameter.
+    List set_of_html_string = 8;
+  }
+}
+
+// Structure containing lists. Note we should fill only
+// one of the fields in this message.
+// We do not add a oneof here because repeated fields cannot be within a oneof.
+message List {
+  repeated string string_list = 1;
+}
+
+// Structure for a number with units object.
+message NumberWithUnits {
+  ObjectType number_type = 1;
+  float real = 2;
+  Fraction fraction = 3;
+  // Controllers are expected to convert the unit into a string
```

Comment:
This is actually a domain object, not a view model object, so we could conceivably have different model objects that contain UI-specific data. We should model these protos based on what we need to store on-disk. If units are already stored as strings in Oppia, then let's keep this.  Otherwise, let's use a type closer to what the backend uses.

Reply:
Sounds good..based on the schema here - https://github.com/oppia/oppia/blob/ae11e9dbeac66b5faeb68b7a2b1f16ccfdbc5ba6/extensions/objects/models/objects.py#L937
I believe there is a string value and an int value

Reply:
I confirm there is indeed a string value and an int value, but note that a "Units" object is a list of (string, int) things.

For example, "kg m s^-2" is represented as [(kg, 1), (m, 1), (s, -2)] (but with dicts instead of tuples.